src/course-search/FacetDropdown.jsx

```diff
@@ -1,33 +1,79 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Dropdown } from '@edx/paragon';
+import { Dropdown, Input } from '@edx/paragon';
 import classNames from 'classnames';
 
-const FacetDropdown = ({ title, items, isBold }) => (
-  <div className="facet-list">
-    <Dropdown className="mb-0 mr-md-3">
-      <Dropdown.Toggle
-        className={
-          classNames(
-            'bg-white', 'text-capitalize', 'rounded-0', 'border-0',
-            'd-flex', 'justify-content-between', 'align-items-center', 'text-dark',
-            { 'font-weight-bold': isBold },
-          )
-        }
-      >
-        {title}
-      </Dropdown.Toggle>
-      <Dropdown.Menu>
-        {items}
-      </Dropdown.Menu>
-    </Dropdown>
-  </div>
-);
+const FacetDropdown = ({
+  title,
+  items,
+  isBold,
+  typeahead,
+  searchForItems,
+}) => {
+  const handleSearch = (value) => {
+    // when user is erasing the input and input is empty we need to reset the filtering
+    if (value.length >= typeahead.minLength || value.length === 0) {
+      searchForItems(value);
+    }
+  };
+
+  return (
+    <div className="facet-list">
+      <Dropdown className={classNames('mb-0 mr-md-3', { typeahead: !!typeahead })}>
+        <Dropdown.Toggle
+          className={
+            classNames(
+              'bg-white', 'text-capitalize', 'rounded-0', 'border-0',
+              'd-flex', 'justify-content-between', 'align-items-center', 'text-dark',
+              { 'font-weight-bold': isBold },
+            )
+          }
+        >
+          {title}
+        </Dropdown.Toggle>
+        <Dropdown.Menu>
+          {
+            typeahead && (
+              <div className="search-input-wrapper">
```

Comment:
I think there's enough difference here that we need two separate components; a TypeaheadFacetDropdown and a FacetDropdown. What they share can go in a `BaseFacetDropdown` component. 
This will keep them easy to reason about and test.

Reply:
Also, can this className go directly on the input instead of wrapping it?

Reply:
Alternately, split off the inner part of this; everything within the DropdownMenu, into a `TypeaheadDropdownMenu` and a `FacetDropdownMenu` (my names are probably not the best here; feel free to do what you want with them!